common/network-common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -117,21 +117,32 @@ public static String genStreamChunkId(long streamId, int chunkId) {
 
   @Override
   public void connectionTerminated(Channel channel) {
+    boolean failedToReleaseBuffers = false;
+
     // Close all streams which have been associated with the channel.
     for (Map.Entry<Long, StreamState> entry: streams.entrySet()) {
       StreamState state = entry.getValue();
       if (state.associatedChannel == channel) {
         streams.remove(entry.getKey());
 
-        // Release all remaining buffers.
-        while (state.buffers.hasNext()) {
-          ManagedBuffer buffer = state.buffers.next();
-          if (buffer != null) {
-            buffer.release();
+        try {
+          // Release all remaining buffers.
+          while (state.buffers.hasNext()) {
+            ManagedBuffer buffer = state.buffers.next();
+            if (buffer != null) {
+              buffer.release();
+            }
           }
+        } catch (RuntimeException e) {
```

Comment:
Is this the only exception type we need to deal with here?

Reply:
Other kind of exception would be Error which `TransportRequestHandler.channelInactive` is also not catching it and let the process fail.